pkg/apis/monitoring/v1/types.go

```diff
@@ -317,6 +317,8 @@ type CommonPrometheusFields struct {
 	// This is an experimental feature, this behaviour could
 	// change or be removed in the future.
 	// Only valid in Prometheus versions 2.28.0 and newer.
+	// Supported units: B, KB, KiB, MB, MiB, GB, GiB, TB, TiB, PB, PiB, EB, EiB Ex: `512MB`.
+	// +kubebuilder:validation:Pattern:="(^0|([0-9]*[.])?[0-9]+((K|M|G|T|E|P)i?)?B)$"
```

Comment:
instead of duplicating the regex here and a few lines below, it would be good to create an intermediate type (`type BytesSize string`) so we don't have to update the expression at several places.